JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt

```diff
@@ -121,103 +139,160 @@ fun HomeScreen(
 fun HomeScreen(
     uiState: HomeUiState,
     onToggleFavorite: (String) -> Unit,
+    onSelectPost: (String) -> Unit,
     onRefreshPosts: () -> Unit,
     onErrorDismiss: (Long) -> Unit,
-    navigateToArticle: (String) -> Unit,
+    onInteractWithList: () -> Unit,
+    onInteractWithDetail: (String) -> Unit,
+    navigateToExpandedArticle: (String) -> Unit,
     openDrawer: () -> Unit,
     scaffoldState: ScaffoldState
 ) {
-    val scrollState = rememberLazyListState()
-    Scaffold(
-        scaffoldState = scaffoldState,
-        snackbarHost = { SnackbarHost(hostState = it, modifier = Modifier.systemBarsPadding()) },
-        topBar = {
-            val title = stringResource(id = R.string.app_name)
-            InsetAwareTopAppBar(
-                title = {
-                    Icon(
-                        painter = painterResource(R.drawable.ic_jetnews_wordmark),
-                        contentDescription = title,
-                        tint = MaterialTheme.colors.onBackground,
-                        modifier = Modifier
-                            .fillMaxSize()
-                            .padding(bottom = 4.dp, top = 10.dp)
+    // Construct the lazy list states for the list and the details outside of deciding which one to show.
+    // This allows the associated state to survive beyond that decision, and therefore we get to preserve the scroll
+    // throughout any changes to the content.
+    val listLazyListState = rememberLazyListState()
+    val detailLazyListStates = uiState.posts.associate { post ->
+        key(post.id) {
+            post.id to rememberLazyListState()
+        }
+    }
+
+    BoxWithConstraints {
+        val useListDetail = when (getWindowSize(maxWidth)) {
+            WindowSize.Compact, WindowSize.Medium -> false
+            WindowSize.Expanded -> true
+        }
+
+        if (useListDetail || uiState.isListOpen) {
```

Comment:
What's this exactly checking? The `else` branch also looks weird just having the `ArticleScreen` when this is supposed to be `HomeScreen`. Is there a better way to handle this? 

Have some problems with this if statement: 1) it's not clear when this should be true or false, and 2) need to figure out if there's a way to remove the `ArticleScreen` call from the else branch

Reply:
I changed this with the latest commit, which more cleanly decides between the 3 possible "screens" from the `HomeRoute` `@Composable`. This also makes it clearer that it's fine to call into the `ArticleScreen`, since the `HomeRoute` is a higher level component.